pkg/cmd/admin/image/verify-signature.go

```diff
@@ -0,0 +1,366 @@
+package image
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"strings"
+
+	"github.com/containers/image/docker/policyconfiguration"
+	"github.com/containers/image/docker/reference"
+	"github.com/containers/image/signature"
+	sigtypes "github.com/containers/image/types"
+	"github.com/openshift/origin/pkg/client"
+	"github.com/openshift/origin/pkg/cmd/templates"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	imageapi "github.com/openshift/origin/pkg/image/api"
+
+	"github.com/spf13/cobra"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+)
+
+var (
+	verifyImageSignatureLongDesc = templates.LongDesc(`
+	Verifies the imported image signature using the local public key.
+
+	This command verifies if the signature attached to an image is trusted by
+	using the provided public GPG key.
+	Trusted image means that the image signature was signed by a valid GPG key and the image identity
+	provided by the signature content matches with the image.
+	By default, this command will not record a signature condition back to the Image object but only
+	print the verification status to the console.
+
+	To record a new condition, you have to pass the "--confirm" flag.
+	`)
+
+	verifyImageSignatureExample = templates.Examples(`
+	# Verify the image signature using the local GNUPG keychan and record the status as a condition to image
+	%[1]s sha256:c841e9b64e4579bd56c794bdd7c36e1c257110fd2404bebbb8b613e4935228c4 --expected-identity=registry.local:5000/foo/bar:v1
+	`)
+)
+
+type VerifyImageSignatureOptions struct {
+	InputImage        string
+	ExpectedIdentity  string
+	PublicKeyFilename string
+	PublicKey         []byte
+	Confirm           bool
+	Remove            bool
+	CurrentUser       string
+
+	Client client.Interface
+	Out    io.Writer
+	ErrOut io.Writer
+}
+
+func NewCmdVerifyImageSignature(name, fullName string, f *clientcmd.Factory, out, errOut io.Writer) *cobra.Command {
+	opts := &VerifyImageSignatureOptions{ErrOut: errOut, Out: out}
+	cmd := &cobra.Command{
+		Use:     fmt.Sprintf("%s IMAGE [--confirm]", name),
```

Comment:
This should mention `--expected-identity=…` since it's mandatory argument.

I'd prefer just `%s IMAGE EXPECTED_IDENTITY [--confirm]`, but I'm repeating myself.